fundamental-concepts/data-structures-stdlib/1_records.md

```diff
@@ -0,0 +1,87 @@
+One of the most basic data types in DAML is the record. Here's an example:
+
+<pre>
+data Address = Address {
+  street: Text,
+  city: Text,
+  country: Text
+}
+</pre>
+
+You define a record by listing its field names together with their types. Records can be nested:
+
+<pre>
+data Person = Person {
+  name: Text,
+  address: Address,
+  age: Int
+}
+</pre>
+
+You can create a new `Person` record by assigning each field with a value:
+
+<pre>
+testPerson: Person
+testPerson = Person {name = "Alice", address = Address {street = "RabbitStreet", city = "QueenOfHearts", country = "Wonderland"}, age = "7"}
+</pre>
+
+When you write a function and you need to **deconstruct** a record you can **pattern match**:
+
+<pre>
+prettyAddress: Address -> Text
+prettyAddress (Address aStreet aCity aCompany) = aStreet <> " - " <> aCity <> " - " <> aCompany
+</pre>
+
+Here we bound the variables `aStreet`, `aCity` and `aCompany` to the fields of the given `Address`
+record in order of its definition.
+
+You can also use the `..` syntax to get the record field names bound to variables of the same name:
+
+<pre>
+prettyAddress: Address -> Text
+prettyAddress Address {..} = street <> " - " <> city <> " - " <> country
+</pre>
+
+Or you can simply use the field accessors:
+
+<pre>
+prettyAddress: Address -> Text
+prettyAddress addr = (street addr) <> " - " <> (city addr) <> " - " <> (country addr)
+</pre>
+
+>> What will the following return: age testPerson? <<
+=== 7
+
+>> What will the following return: prettyAddress $ address testPerson? <<
+===  RabbitStreet - QueenOfHearts - Wonderland
```

Comment:
Would move this to the end of the section after the user has actually run the script

Reply:
I'd leave it here, so the user gets quizzed right after the concepts where introduced.